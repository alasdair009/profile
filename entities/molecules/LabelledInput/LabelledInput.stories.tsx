import { LabelledInput } from "./LabelledInput";
import { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof LabelledInput> = {
  component: LabelledInput,
};
export default meta;

export const Text: StoryObj<typeof LabelledInput> = {
  args: {
    label: "Text entry",
    type: "text",
  },
};

export const Email: StoryObj<typeof LabelledInput> = {
  args: {
    label: "Email entry",
    type: "email",
  },
};

export const Passowrd: StoryObj<typeof LabelledInput> = {
  args: {
    label: "Password entry",
    type: "password",
  },
};

export const Required: StoryObj<typeof LabelledInput> = {
  args: {
    label: "Required entry",
    required: true,
  },
};

export const IsInvalid: StoryObj<typeof LabelledInput> = {
  args: {
    label: "isInvalid entry",
    required: true,
    isInvalid: true,
  },
};
