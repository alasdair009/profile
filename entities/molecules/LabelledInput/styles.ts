"use client";
import styled from "styled-components";
import { globalInputMaxWidth, sizes } from "@/entities";
import { rem } from "polished";

export const Root = styled.label`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto ${sizes.s8.rem};
  max-width: ${rem(globalInputMaxWidth)};
  width: 100%;
`;

export const LabelText = styled.span`
  display: block;
  max-width: ${rem(globalInputMaxWidth)};
  text-align: center;
  width: 100%;
`;
