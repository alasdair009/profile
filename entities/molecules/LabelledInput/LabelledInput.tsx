import { LabelText, Root } from "./styles";
import { HTMLAttributes } from "react";
import { TextInputType } from "../../atoms/BaseInput/BaseInput.types";
import { BaseInput, ErrorText } from "@/entities";

type LabelledInputProps = {
  label: string;
  type: TextInputType;
  name: string;
  value?: string;
  required?: boolean;
  isInvalid?: boolean;
  errorText?: string;
} & HTMLAttributes<HTMLInputElement>;

/**
 * Base input type
 */
export function LabelledInput({
  isInvalid,
  type,
  name,
  label,
  errorText = `Please enter a valid ${label}`,
  ...rest
}: LabelledInputProps) {
  return (
    <Root>
      <LabelText>{label}</LabelText>
      <BaseInput type={type} isInvalid={isInvalid} {...rest} />
      <ErrorText shown={isInvalid || false}>{errorText}</ErrorText>
    </Root>
  );
}
