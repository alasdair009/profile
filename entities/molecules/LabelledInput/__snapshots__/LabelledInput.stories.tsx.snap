// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/LabelledInput Email play-test 1`] = `
<label data-testid="LabelledInput"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Email entry
  </span>
  <input data-testid="BaseInput"
         required
         class="sc-elDIKY fBsEuy"
         type="email"
         value="hello"
  >
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;

exports[`molecules/LabelledInput IsInvalid play-test 1`] = `
<label data-testid="LabelledInput"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    isInvalid entry
  </span>
  <input data-testid="BaseInput"
         required
         class="sc-elDIKY gPWKCy"
         type="text"
         value="hello"
  >
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH cfxBNa"
  >
  </span>
</label>
`;

exports[`molecules/LabelledInput Password play-test 1`] = `
<label data-testid="LabelledInput"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Password entry
  </span>
  <input data-testid="BaseInput"
         required
         class="sc-elDIKY fBsEuy"
         type="password"
         value="hello"
  >
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;

exports[`molecules/LabelledInput Required play-test 1`] = `
<label data-testid="LabelledInput"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Required entry
  </span>
  <input data-testid="BaseInput"
         required
         class="sc-elDIKY fBsEuy"
         type="text"
         value="hello"
  >
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;

exports[`molecules/LabelledInput Text play-test 1`] = `
<label data-testid="LabelledInput"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Text entry
  </span>
  <input data-testid="BaseInput"
         required
         class="sc-elDIKY fBsEuy"
         type="text"
         value="hello"
  >
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;
